spec/controllers/v0/facilities/ccp_controller_spec.rb

```diff
@@ -3,35 +3,256 @@
 require 'rails_helper'
 
 RSpec.describe V0::Facilities::CcpController, type: :controller do
-  regex_matcher = lambda { |r1, r2|
-    r1.uri.match(r2.uri)
-  }
-
-  it 'has a certain shape' do
-    VCR.use_cassette('facilities/va/ppms', match_requests_on: [regex_matcher]) do
-      get 'show', params: { id: 'ccp_12345' }
-      expect(response).to have_http_status(:ok)
-      bod = JSON.parse(response.body)
-      expect(bod['data']['id']).to include('ccp_')
-      expect(bod['data']['type']).to eq('cc_provider')
+  def fake_address(attributes)
+    if attributes.slice(
+      :AddressStreet,
+      :AddressCity,
+      :AddressStateProvince,
+      :AddressPostalCode
+    ).values.all?
+      {
+        'street' => provider.AddressStreet,
+        'city' => provider.AddressCity,
+        'state' => provider.AddressStateProvince,
+        'zip' => provider.AddressPostalCode
+      }
+    else
+      {}
+    end
+  end
+
+  def fake_data(attributes)
+    {
+      'attributes' => {
+        'acc_new_patients' => attributes[:IsAcceptingNewPatients],
+        'address' => fake_address(attributes),
+        'caresite_phone' => attributes[:CareSitePhoneNumber],
+        'email' => attributes[:Email],
+        'fax' => attributes[:OrganizationFax],
+        'gender' => attributes[:ProviderGender],
+        'lat' => attributes[:Latitude],
+        'long' => attributes[:Longitude],
+        'name' => attributes[:Name],
+        'phone' => attributes[:MainPhone],
+        'pref_contact' => attributes[:ContactMethod],
+        'specialty' => [],
+        'unique_id' => attributes[:ProviderIdentifier]
+      },
+      'id' => "ccp_#{attributes[:ProviderIdentifier]}",
+      'type' => 'cc_provider'
+    }
+  end
+
+  def fake_provider_serializer(provider_hash, details_hash = {}, set = true)
+    attributes = details_hash.merge(provider_hash)
+    data = fake_data(attributes)
+    if set
+      { 'data' => [data] }
+    else
+      { 'data' => data }
+    end
+  end
```

Comment:
Shouldn't the response you're creating from the controller action already handle all of this?

Reply:
I split these into two methods